### DISCUSSION TITLE
Running a function inside exports.config file

### DISCUSSION THREAD

**Discussion by jurjandrei** [webdriverio/webdriverio#12004] [Q&A] [answered]
Hello,

I have a function which parses suites.js file and retrieves the paths in an array to the spec files which are about to run. The mentioned function is exported at the end of the file and the imported in wdio.conf.js
Is there a possibility to call the function under export.config = {
  suites: {
    mySuite: callMyFunction(),
  },
}
THank you

**Comment by christian-bromann** [comment]
Please describe what you are trying to achieve and what error you are seeing?

**Reply by jurjandrei** [reply]
@christian-bromann so I want to generate dynamically the suites with paths to the spec files which are added in the exports.config = { suites: [] }

So for example I have this function:
```
const testResults = async () => {
  const testPlatform = Object.entries(Tests)
    .filter(([key, value]) => value.COUNTRY_RUNTIME.includes(process.env.COUNTRY.toLowerCase())
          && value.STATES_RUNTIME.includes(process.env.STATE.toLowerCase())
          && value.BRAND_RUNTIME.includes(process.env.BRAND.toLowerCase())
          && value.SUITE_RUNTIME.includes(process.env.SUITE.toLowerCase())
          && value.ENVIRONMENT_RUNTIME.includes(process.env.ENV.toLowerCase())
          && value.PLATFORM_RUNTIME.includes(process.env.PLATFORM.toLowerCase()))
    .map(([key, value]) => value.RELATIVE_PATH);
  return testPlatform;
```
};
Since this function returns an array with the relative paths to the spec files, then I want to use this function inside wdio.conf.js file in when giving the tests in suites: {}
Something like
```
exports.config = {
  updateJob: false,
  user: userName,
  key: accessKey,
  specs: [
    './test/specs/**/*.spec.js',
  ],
  exclude: [],
  maxInstances: parseInt(process.env.MAX_INSTANCES),
  capabilities: [Capability],
  suites: {
    testResults,
  },
}
```

The error received is: Error: The suite(s) "smoke" you specified don't exist in your config file or doesn't contain any files!

even if console.log(testPlatform) in the above function prints the relative paths as an array

**Reply by christian-bromann** [reply, marked as answer]
You define `testResults` which is an async function and apply it to `suites` like this:

```ts
  suites: {
    testResults,
  },
```

which is incorrect because it doesn't evaluate functions. What you want to do is remove the `async` and define it as an object:

```ts
const testResults = Object.entries(Tests)
  .filter(([key, value]) => value.COUNTRY_RUNTIME.includes(process.env.COUNTRY.toLowerCase())
        && value.STATES_RUNTIME.includes(process.env.STATE.toLowerCase())
        && value.BRAND_RUNTIME.includes(process.env.BRAND.toLowerCase())
        && value.SUITE_RUNTIME.includes(process.env.SUITE.toLowerCase())
        && value.ENVIRONMENT_RUNTIME.includes(process.env.ENV.toLowerCase())
        && value.PLATFORM_RUNTIME.includes(process.env.PLATFORM.toLowerCase()))
  .map(([key, value]) => value.RELATIVE_PATH);
```
